src/main/scala/treadle/vcd/VCD.scala

```diff
@@ -10,10 +10,18 @@ import logger.LazyLogging
 import collection._
 import java.util.{Date, TimeZone}
 
+import firrtl.FileUtils
+import firrtl.options.StageMain
+
 import scala.collection.mutable.ArrayBuffer
 import scala.util.matching.Regex
 
-object VCD extends LazyLogging {
+/**
+  * This effective stage main exercises vcd reading and optionally writing
+  * and depending up filtering options can pull out only those change values that
+  * are specific to a particular module
+  */
+object VCD extends StageMain(new VcdStage) with LazyLogging {
```

Comment:
@chick Why does the VCD object have to extend `StageMain`?
I saw that you added a `VCDStage` already. What is the difference?

Reply:
Could we move this into a separate object?
Like maybe just: `object VCDStageMain extends StageMain(new VcdStage) {}`?

Reply:
Then also move it into `VcdStage.scala`.

Reply:
@ekiwi afaik it's just a pattern. `StageMain` provides a `main` that parses the CLI args and calls the associated stage's `execute` method. The pattern would ordinarily have a `VcdMain` but the existing API is to just to have the object`VCD` have the `main`. I guess I could create a `VcdMain` as the new canonical way of calling it. I'm not sure how I would deprecate the existing `VCD#main`. Do you think it's worth doing that.

Reply:
Oh, I see. So this is done for backwards compatibility reasons. That sounds OK!